flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -0,0 +1,291 @@
+import org.apache.flink.configuration.CoreOptions;
+import org.apache.flink.kubernetes.configuration.KubernetesConfigOptions;
+import org.apache.flink.runtime.clusterframework.BootstrapTools;
+import org.apache.flink.runtime.clusterframework.ContaineredTaskManagerParameters;
+
+import io.fabric8.kubernetes.api.model.ConfigMapVolumeSourceBuilder;
+import io.fabric8.kubernetes.api.model.KeyToPath;
+import io.fabric8.kubernetes.api.model.Quantity;
+import io.fabric8.kubernetes.api.model.ResourceRequirements;
+import io.fabric8.kubernetes.api.model.ResourceRequirementsBuilder;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.fabric8.kubernetes.api.model.VolumeMountBuilder;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nullable;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.configuration.GlobalConfiguration.FLINK_CONF_FILENAME;
+import static org.apache.flink.kubernetes.utils.Constants.CONFIG_FILE_LOG4J_NAME;
+import static org.apache.flink.kubernetes.utils.Constants.CONFIG_FILE_LOGBACK_NAME;
+import static org.apache.flink.kubernetes.utils.Constants.CONFIG_MAP_PREFIX;
+import static org.apache.flink.kubernetes.utils.Constants.FLINK_CONF_VOLUME;
+
+/**
+ * Common utils for Kubernetes.
+ */
+public class KubernetesUtils {
+
+	/**
+	 * Read file content to string.
+	 * @param filePath file path
+	 * @return content
+	 */
+	public static String getContentFromFile(String filePath) throws FileNotFoundException {
+		File file = new File(filePath);
+		if (file.exists()) {
+			StringBuilder content = new StringBuilder();
+			String line;
+			try (BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(file)))){
+				while ((line = reader.readLine()) != null) {
+					content.append(line).append(System.lineSeparator());
+				}
+			} catch (IOException e) {
+				throw new RuntimeException("Error read file content.", e);
+			}
+			return content.toString();
+		}
+		throw new FileNotFoundException("File " + filePath + " not exists.");
+	}
+
+	/**
+	 * Generates the shell command to start a job manager for kubernetes.
+	 * @param flinkConfig The Flink configuration.
+	 * @param jobManagerMemoryMb JobManager heap size.
+	 * @param configDirectory The configuration directory for the flink-conf.yaml
+	 * @param logDirectory The log directory.
+	 * @param hasLogback Uses logback?
+	 * @param hasLog4j Uses log4j?
+	 * @param mainClass The main class to start with.
+	 * @param mainArgs The args for main class.
+	 * @return A String containing the job manager startup command.
+	 */
+	public static String getJobManagerStartCommand(
```

Comment:
`getJobManagerStartCommand` and `getTaskManagerShellCommand ` look quite similar except for a few lines or arguments. Could we extract the common logics into one method to avoid code duplication?

Reply:
I will try to extract more common codes to a separate method.